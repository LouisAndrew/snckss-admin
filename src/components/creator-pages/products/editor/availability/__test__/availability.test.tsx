import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import Availabilty from '..'

describe('Availability component on product editor', () => {
        let valueContainer: boolean = true

        const mockHeaderText: string = 'Input'
        const mockIsAvailable: boolean = true
        const mockHandleChange = jest.fn(
                (event: React.ChangeEvent<HTMLInputElement>) => {
                        valueContainer = event.target.checked
                }
        )

        afterEach(cleanup)
        const el: React.ReactElement = (
                <Availabilty
                        isAvailable={mockIsAvailable}
                        headerText={mockHeaderText}
                        handleChange={mockHandleChange}
                />
        )

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        it('should render with label text correctly', () => {
                const { getByLabelText } = render(el)
                const input: HTMLInputElement = getByLabelText(
                        mockHeaderText
                ) as HTMLInputElement

                expect(input).toBeInTheDocument()
        })

        it('should render with default value correctly', () => {
                const { getByLabelText } = render(el)
                const input: HTMLInputElement = getByLabelText(
                        mockHeaderText
                ) as HTMLInputElement

                expect(input.checked).toBe(mockIsAvailable)
        })

        it('should render and react to user input correctly', () => {
                const { getByLabelText } = render(el)
                const input: HTMLInputElement = getByLabelText(
                        mockHeaderText
                ) as HTMLInputElement

                const changeFromUser: boolean = false
                fireEvent.click(input)

                expect(mockHandleChange).toBeCalled()
                expect(valueContainer).toBe(changeFromUser)
        })

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
