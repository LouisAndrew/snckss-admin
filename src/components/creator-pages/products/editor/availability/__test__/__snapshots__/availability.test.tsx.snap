// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Availability component on product editor matches snapshot 1`] = `
<div
  className="available form-group"
>
  <label
    className=""
    htmlFor="available"
  >
    Input
  </label>
  <input
    checked={true}
    className="form-check-input"
    id="available"
    name="available"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            SyntheticEvent {
              "_dispatchInstances": null,
              "_dispatchListeners": null,
              "_targetInst": null,
              "bubbles": null,
              "cancelable": null,
              "currentTarget": [Function],
              "defaultPrevented": null,
              "dispatchConfig": null,
              "eventPhase": null,
              "isDefaultPrevented": [Function],
              "isPropagationStopped": [Function],
              "isTrusted": null,
              "nativeEvent": null,
              "target": null,
              "timeStamp": [Function],
              "type": null,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    type="checkbox"
  />
</div>
`;
